test/git_environment.go

```diff
@@ -44,37 +49,100 @@ func NewGitEnvironment(baseDir string) (*GitEnvironment, error) {
 
 // NewStandardGitEnvironment provides a GitEnvironment in the given directory,
 // fully populated as a standardized setup for scenarios.
-func NewStandardGitEnvironment(dir string) (result *GitEnvironment, err error) {
-	result, err = NewGitEnvironment(dir)
+func NewStandardGitEnvironment(dir string) (gitEnv *GitEnvironment, err error) {
+	gitEnv, err = NewGitEnvironment(dir)
 	if err != nil {
-		return result, errors.Wrapf(err, "cannot create a new standard environment")
+		return gitEnv, errors.Wrapf(err, "cannot create a new standard environment")
 	}
 
 	// create the origin repo
-	result.OriginRepo, err = InitGitRepository(result.originRepoPath(), true)
+	gitEnv.OriginRepo, err = InitGitRepository(gitEnv.originRepoPath())
```

Comment:
I don't use a bare repo for origin anymore but a "normal" one instead. I leave the `master` branch checked out in it since we don't use it. This allows pushing to it as if it were a bare repo while it simplifies checking the commits in it: I can query the origin repo directly now instead of having to clone it and query the clone, as our old Ruby specs did.